### ISSUE TITLE
[FR] Quick-search suppl/mfg site for SKU/MPN

### ISSUE THREAD

**Issue by LavissaWoW** [inventree/InvenTree#4606]
### Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find a similar feature request

### Problem statement

Registering a supplier part involves adding both part information, descriptions and links in order to make ordering as easy as possible. There is as of yet no quick copy button for SKUs in neither PO line items nor supplier parts on part views. This option exists solely on the Supplier Part Detail view. When executing PO's, this produces a lot of extra clicks or awkward click-and-drag marking and copying of SKUs. 

### Suggested solution

As a default, expand the quick copy button/icon to line items and part details. Moving on, it's possible to setup a generated search link for the relevant supplier. When creating a supplier part, the SKU is set as a required field. With some small additions to the company model, you have everything you need to enable a quick-search button to appear behind the given SKU.

Diving a bit into the matter: 
Say you have Mouser as a supplier. Including a new field on the model for a "search link", you can input "https://mouser.com/c/?q=" as the URL prefix for searching. 
Given a supplier part from Mouser with i.e. the SKU "139-ZX60-P33ULN" you can then present a button (i.e. FontAwesome icon) which opens a new tab with the complete URL "https://mouser.com/c/?q=139-ZX60-P33ULN"

As a general priority list for which attribute to display:
1. Supplier part Link
2. Autogenerated search link based on SKU
3. Copy SKU button

This makes it so that the presence of a link will override the generated search link in all cases. This would be important in cases where a search does not send you to a unique product page. Like when your SKU has variations on the supplier site with prefixes/affixes and so on. 

### Describe alternatives you've considered

Opening line items (PO) and supplier parts (Part details) in new tabs to copy SKU manually.

### Do you want to develop this?

- [X] I want to develop this.

**Comment by SchrodingersGat** [comment]
Something specific to a particular supplier like this would have to be a plugin. Currently we do not have a plugin entry point which would provide this functionality, but perhaps there's a generic way to hook in?

**Comment by LavissaWoW** [comment]
The functionality isn't strictly unique to a given supplier per se. It's generic URL concatenation. You just need a place to store the URL pattern itself. Expanding the Company model with a new field, the same logic can be applied to each supplier you have where you know their search URL. 

Digi-Key as an example uses: https://www.digikey.com/en/products/result?keywords=abcd
Farnell: https://uk.farnell.com/w/search/prl/results?st=abcd

And so on. 
Granted, I get this might be more of an "advanced" use case. However, it's just a link like how one links to the item itself using the link field. This link is simply a concatenation of (i.e.) the above and a given SKU.

**Comment by SchrodingersGat** [comment]
@LavissaWoW I guess that this is a pretty reasonable request - if you are happy to submit a PR, I'm happy to review!

**Comment by matmair** [comment]
Changing the URLs needs to be restricted to superusers and the link needs to render with [noreferrer](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel/noreferrer) to avoid making the system vulnerable and leaking sensitive information

**Comment by SchrodingersGat** [comment]
Great point @matmair regarding security issues!

**Comment by LavissaWoW** [comment]
Was considering maybe splitting this into 2-3 merge requests to keep commit history clear and understandable.

- [ ] Adding clipboard icons to dynamic tables with SKU and MPN
- [ ] Making external-pointing links open in New tabs 
- [ ] Implementing search patterns as described above
- [ ] Setting up dynamic priorities for link fields following the above list

Let me know if this list is a nay. If no nays, I'll proceed with this thought.

**Comment by matmair** [comment]
@LavissaWoW do you have a working PoC? Please make sure the desired outcome works before we merge things that do not lead to the desired outcome in the end. I am not really sure it makes sense to split up the FR as the targets themself seem to be only a few LoC each.
The only one I would split out is the last - still not sure if that feature will not mostly lead to confusion.

Please consider the added space needed for icons in the tables as it will consume a lot of limited screen space.

**Comment by LavissaWoW** [comment]
@matmair I've got a working PoC for the first two items. 
At the moment, it appears that the icons aren't causing any crowding issues.
There is ofc. an argument to be made to truncate link displays in some table views to get some real-estate back.

I filled the mfg/suppl table values with max lengths, and this is how that looks on a 21:9 display. Force some linebreaks on MPNs/SKUs without natural breakpoints, and you're reducing a lot of risk for these occupying a lot of width.

![image](https://user-images.githubusercontent.com/7506688/233807320-3a08678f-fb9b-4c27-9ed2-a668a2c3f5b8.png)
